lib/Reconciler.lua

```diff
@@ -425,11 +419,24 @@ end
 	element, created using debug.traceback(), that points to where the element
 	was created.
 ]]
-function Reconciler._setRbxProp(rbx, key, value)
+function Reconciler._setRbxProp(rbx, key, value, element)
 	if type(key) == "string" then
 		-- Regular property
 
-		rbx[key] = value
+		local success, err = pcall(_rawSet, rbx, key, value)
+
+		if not success then
+			local source = element.source or "\n\t<Use Roact.DEBUG_ENABLE() to enable detailed tracebacks>\n"
+
+			local message = ("Failed to set property %s on primitive instance of class %s\n%s\n%s"):format(
+				key,
+				rbx.ClassName,
+				err,
+				source
+			)
+
+			error(message, 0)
+		end
```

Comment:
Should we introduce nice stack trace logic on the error cases below as well?

I haven't seen them ever come up in real code, but giving them equally pleasant messages would be good. I can understand if you'd want to push that into a different PR as well, though.